drivers/i2c/i2c_mchp_xec.c

```diff
@@ -371,39 +427,180 @@ static int i2c_xec_transfer(const struct device *dev, struct i2c_msg *msgs,
 {
 	int ret = 0;
 
+#ifdef CONFIG_I2C_SLAVE
+	struct i2c_xec_data *data = dev->driver_data;
+
+	if (data->slave_attached) {
+		LOG_ERR("Device is registered as slave");
+		return -EBUSY;
+	}
+#endif
+
 	addr <<= 1;
 	for (int i = 0U; i < num_msgs; i++) {
 		if ((msgs[i].flags & I2C_MSG_RW_MASK) == I2C_MSG_WRITE) {
 			ret = i2c_xec_poll_write(dev, msgs[i], addr);
 			if (ret) {
+				LOG_ERR("Write error: %d", ret);
 				return ret;
 			}
 		} else {
 			ret = i2c_xec_poll_read(dev, msgs[i], addr);
+			if (ret) {
+				LOG_ERR("Read error: %d", ret);
+				return ret;
+			}
+		}
+	}
+
+	return 0;
+}
+
+static void i2c_xec_bus_isr(void *arg)
+{
+#ifdef CONFIG_I2C_SLAVE
+	struct device *dev = (struct device *)arg;
+	const struct i2c_xec_config *config =
+		(const struct i2c_xec_config *const) (dev->config_info);
+	struct i2c_xec_data *data = dev->driver_data;
+	const struct i2c_slave_callbacks *slave_cb = data->slave_cfg->callbacks;
+	uint32_t ba = config->base_addr;
+
+	uint32_t status;
+	uint8_t val;
+
+	uint8_t dummy = 0U;
+
+	if (!data->slave_attached) {
+		return;
+	}
+
+	/* Get current status */
+	status = MCHP_I2C_SMB_STS_RO(ba);
+
+	/* Bus Error */
+	if (status & MCHP_I2C_SMB_STS_BER) {
+		if (slave_cb->stop != NULL) {
+			slave_cb->stop(data->slave_cfg);
+		}
+		restart_slave(ba);
+		goto clear_iag;
+	}
+
+	/* External stop */
+	if (status & MCHP_I2C_SMB_STS_EXT_STOP) {
+		if (slave_cb->stop != NULL) {
+			slave_cb->stop(data->slave_cfg);
+		}
+		dummy = MCHP_I2C_SMB_DATA(ba);
+		goto clear_iag;
+	}
+
+	if ((!(status & MCHP_I2C_SMB_STS_PIN)) &&
```

Comment:
The HW can assert AAS first if an address match then PIN after the 9th clock. If you have AAS interrupt enabled you may enter the ISR with AAS==1 and PIN==1. For this case line 499 is false and the ISR runs the next if-else to write or read data.  Can you instrument the ISR to check if AAS==1 and PIN==1 at line 499 when the problem occurs?